### PR TITLE
Solutions for module-1, module-2

### PR DESCRIPTION
**Pull request by vmashnitskaya**
Hello @szikszail, please review the solutions for the first module.

### CODE REVIEW

File: module-1/classification.js [hunk 1]
```diff
@@ -23,7 +23,21 @@ function grade(score) {
      * Also take into consideration the documentation of the function!
      */
     // PLACE YOUR CODE BETWEEN THIS...
-
+if(score >= 0 && score <= 100){
```
Comment (szikszail) [inline module-1/classification.js]: Could you please format it a bit better to make it more readable?

**Review by szikszail — CHANGES_REQUESTED** [review]
Seems to be good solutions, but without formatting it is not too readable. Could you please format your code? In VSCode, it is `Alt + Shift + F`

File: module-1/classification.js (hunk 1 above)
Comment (vmashnitskaya) [reply module-1/classification.js]: Thank you for the point. I have solved the issue.

**Review by szikszail — COMMENTED** [review]
Nice solutions! I am waiting for further implementations.

**Comment by vmashnitskaya** [comment]
Hello @szikszail, could you please help me with the task in Element module? I am unaware how to continue. 
I am also unaware why my tests on addChildren method in Layout class do not pass although I have extended it from Element class.
Thank you!

**Comment by vmashnitskaya** [comment]
No problem with addChildren method in Layout class after Element module import.

File: module-3/Element.js [hunk 2]
```diff
@@ -1 +1,30 @@
...
+        this.children[child.name] = child;
+    }
+    get(name) {
+        if (!name) {
+            return this.element(this.locator);
+        }
+        if (this.children.hasOwnProperty(child.name)) {
+            return this.children[child.name].get();
+        } else {
+            throw new Error('No child with such a name')
```
Comment (szikszail) [inline module-3/Element.js]: What if a child has a child named as the parameter? :)

File: module-3/Element.js [hunk 3]
```diff
@@ -1 +1,30 @@
...
+    }
+    addChildren(child) {
+        if (this.children.hasOwnProperty(child.name)) {
+            throw new Error(child.name + ' is already added.');
+        }
+        this.children[child.name] = child;
+    }
+    get(name) {
+        if (!name) {
+            return this.element(this.locator);
```
Comment (szikszail) [inline module-3/Element.js]: I see your intention. It is actually a good idea to base such method which is Protractor's element method.

**Review by szikszail — CHANGES_REQUESTED** [review]
Please check my comments. I am excited to see the rest of the implementation :)

File: module-3/Element.js (hunk 3 above)
Comment (vmashnitskaya) [reply module-3/Element.js]: Now, with the updates all tests are passed. Thank you!

File: module-1/classification.js [hunk 4]
```diff
@@ -23,7 +23,22 @@ function grade(score) {
...
+            gradeOfStudent = 3;
+        } else if (score >= 60) {
+            gradeOfStudent = 2;
+        } else {
+            gradeOfStudent = 1;
+        }
+    } else {
+        gradeOfStudent = 0;
+    }*/
+    if (score >= 0 && score <= 100) { gradeOfStudent = Math.min(Math.max(Math.ceil((score - 49) / 10), 1), 5) } else { gradeOfStudent = 0 };
```
Comment (szikszail) [inline module-1/classification.js:41]: Almost :) You can also use [conditional operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)

File: module-3/Element.js [hunk 5]
```diff
@@ -1 +1,28 @@
...
+            throw new Error(' Child has already been added.');
+        }
+        this.children[child.name] = child;
+    }
+    get(name) {
+        if (!name) {
+            return element(this.locator);
+        } else if (this.children.hasOwnProperty(name)) {
+            return element(this.children[name].locator);
+        }
```
Comment (szikszail) [inline module-3/Element.js:23]: What if any children have a child named like the parameter?

File: module-3/Elements.js [hunk 6]
```diff
@@ -1 +1,19 @@
...
+        this.children = null;
+    }
+    addChildren() {
+        throw new Error('Elements cannot have children!');
+    }
+    all() {
+        return element.all(this.locator);
+    }
+    get(num) {
+        return num;
```
Comment (szikszail) [inline module-3/Elements.js:16]: Here [this](https://www.protractortest.org/#/api?view=ElementArrayFinder.prototype.get) would be the right function to use.

File: module-3/HomePage.js [hunk 7]
```diff
@@ -1 +1,8 @@
 'use strict';
+const Element = require('../Element');
+const Elements = require('../Elements');
+const Layout = require('../Layout');
+
+class HomePage extends Layout, Element{
```
Comment (szikszail) [inline module-3/HomePage.js:6]: I can and is should only extend one class. Here `Layout` would be the right one

**Review by szikszail — CHANGES_REQUESTED** [review]

**Comment by szikszail** [comment]
Thank you for your solution!

**Pull request closed without merging**